src/models/oracle/facade.js

```diff
@@ -102,6 +102,15 @@ exports.oracleRequest = async (headers, method, params = {}, query = {}, payload
     return await request.sendRequest(url, headers, headers[Enums.Http.Headers.FSPIOP.SOURCE], headers[Enums.Http.Headers.FSPIOP.DESTINATION] || Enums.Http.Headers.FSPIOP.SWITCH.value, method.toUpperCase(), payload || undefined)
   } catch (err) {
     Logger.error(err)
+    // If the error was a 400 from the Oracle, we'll modify the error to generate a response to the
+    // initiator of the request.
+    if (
+      err.name === 'FSPIOPError' &&
+      err.apiErrorCode.code === ErrorHandler.Enums.FSPIOPErrorCodes.DESTINATION_COMMUNICATION_ERROR.code &&
```

Comment:
There is a validate FSPIOPError function

Reply:
Hey, do you mean this one?
https://github.com/mojaloop/central-services-error-handling/blob/master/src/factory.js#L328
It's not a replacement for the functionality here, AFAICT.

Reply:
Ok thats cool, do you think that there is a need to have this in the error handling library? like a factory function that can be refused elsewhere?

Reply:
Tricky question. I guess you're comparing this:
```
    err.name === 'FSPIOPError' &&
    err.apiErrorCode.code === ErrorHandler.Enums.FSPIOPErrorCodes.DESTINATION_COMMUNICATION_ERROR.code
```
with this:
```
    ErrorHandler.errIsFSPIOP(err, ErrorHandler.Enums.FSPIOPErrorCodes.DESTINATION_COMMUNICATION_ERROR)
```
(presumably with an optional second parameter)
My personal preference is the former- as a rule I don't really like to hide a couple of lines behind a function. But the function does cleanly encapsulate a concept, and mean the user doesn't have to know about the implementation. Not the hill I'd die on- I don't _really_ mind either way. Your thoughts?

Reply:
Although, the user does need to know about the implementation a little bit on the very next line. So I'd fall a little more firmly on the side of not building the abstraction.

Reply:
lol i think we are overthinking, lets go with your first option and if a change is need we can look at that later